src/EUTelGeometryTelescopeGeoDescription.cpp

```diff
@@ -883,14 +941,105 @@ const TGeoHMatrix* EUTelGeometryTelescopeGeoDescription::getHMatrix( const doubl
+               streamlog_out(DEBUG5) << "INCREASE TO MINIMUM DISTANCE!" << std::endl;
+                snext = delta;
+                double pt[3];
+                memcpy( pt, gGeoManager->GetCurrentPoint(), 3 * sizeof (double) ); //Get global position
+                const double *dir = gGeoManager->GetCurrentDirection();//Direction vector
+                for ( Int_t i = 0; i < 3; i++ ) pt[i] += delta * dir[i]; //Move the current point slightly in the direction of motion. 
+                nextnode = gGeoManager->FindNode( pt[0], pt[1], pt[2] );//Move to new node where we will begin to look for more radiation length   
+                rad=lastrad*snext; //Calculate radiation length for the increased block.
+                blockEnd += snext;
+           }else{
+                streamlog_out(DEBUG5) << "OVERMAX!" << std::endl;
+                blockEnd += snext;
+                rad = lastrad*snext; //This is the calculated (rad per distance x distance)
+           }
+            total = total + rad;
+            streamlog_out(DEBUG5) << "NEW BLOCK:SensorID: " << sensorID   <<" Block width: " << snext << " Radiation total/per unit length: " << rad<<"/"<< 1.0/lastrad << " Block end position: " << blockEnd  << std::endl;
+            streamlog_out(DEBUG5) << "DECISION: Where should block be placed?"  << std::endl;
+
+            //Now we have the block. We place it in the planes or in the air if excluded.
+            //Work Flow: Check we are at end. If not then set radiation length to sensor if included.Else attach the radiation length last one included and found. 
+            if(sensorID != geo::gGeometry().sensorZOrderToIDWithoutExcludedPlanes().at( geo::gGeometry().sensorZOrderToIDWithoutExcludedPlanes().size()-1 )){
+                if(geo::gGeometry().sensorIDToZOrderWithoutExcludedPlanes().find(sensorID) !=  geo::gGeometry().sensorIDToZOrderWithoutExcludedPlanes().end()){
+                    sensors[sensorID] = sensors[sensorID] +  rad;
+                    sensorLeftSide =sensorID;
+                    air[sensorID] = 0 ;
+                    streamlog_out( DEBUG5 ) << "Placed in sensor:: " << sensorID <<std::endl;
+                }else{
+                    air[sensorLeftSide] = air[sensorLeftSide] +  rad;
+                    streamlog_out( DEBUG5 ) << "Placed in air after sensor:: " << sensorLeftSide <<std::endl;
+                }
+            }else{//IF WE HAVE REACHED THE FINAL SENSOR THEN DO NOT LOOK FOR ANYMORE RADIATION LENGTH
+                sensors[sensorID] = sensors[sensorID] + rad;
+                streamlog_out( DEBUG5 ) << "FINISHED: Sensor: " <<sensorID << " Total Rad: " << total <<std::endl;
+                return total;
+            }
+        }
+    }
+}
+float EUTelGeometryTelescopeGeoDescription::findRadLengthIntegral( const double globalPosStart[], const double globalPosFinish[], bool skipBoundaryPonitsVolumes, 	std::map< const int, double> &sensors, 	std::map< const int, double> &air ) {
```

Comment:
This has been removed. This was the old way of doing this which was rather inelegant.